### PR TITLE
Build all targets in all platforms.

### LINKED ISSUE #154: Build all TFMs on Windows, macOS and Linux
Following this [comment](https://github.com/open-telemetry/opentelemetry-dotnet/pull/151#issuecomment-511596887).

Make `dotnet build` build all targets (currently `net45` and `netstandard2.0`) on Windows, macOS and Linux. [This allows development all 3 platforms without reaching those `#ifdef` that exclude full framework code paths on non-Windows machines.](https://github.com/open-telemetry/opentelemetry-dotnet/pull/151#issuecomment-511592840)

### PR DESCRIPTION
**Pull request by bruno-garcia**
Resolves #154

tldr: Two things changed:
1. Class libraries build all targets on all platforms.
2. Test projects only include full framework targets if OS = Windows.

Mono is not required to be installed since we're using `Microsoft.NETFramework.ReferenceAssemblies` to compile against. 
To run the tests on `net4x` though we'd need to make sure `Mono` is installed but I've tried running it and some tests fail due to some `AppDomain` and `Remoting` usage which I assume isn't there on `Mono`.

The opt-in for `net4x` on Windows approach is better so we don't need to keep the targets we want to compile in all platforms duplicated.

### CODE REVIEW

**Comment by bruno-garcia** [comment]
@lmolkova I'm not sure why the build is failing. Just says: 
image

Any idea what could have gone wrong there?

**Comment by austinlparker** [comment]
Looks like it's something with the `reportgenerator` tool not running properly.

**Comment by bruno-garcia** [comment]
The change didn't really affect where tests run though. Nor what platforms (net4x on *nix) we compile the test projects which makes me think the build is flaky.

All I did was make sure contracts for `net4x` are validated when compiling on non-Windows but perhaps I'm missing something.
 
Please advise.

**Comment by bruno-garcia** [comment]
Now that jaeger exporter was merged in, I've also change the class library and tests to match the rest of the solution.

**Comment by bruno-garcia** [comment]
[Build is green](https://dev.azure.com/opentelemetry/pipelines/_build/results?buildId=306) but GitHub status check is hanging on _running_

**Comment by lmolkova** [comment]
> [Build is green](https://dev.azure.com/opentelemetry/pipelines/_build/results?buildId=306) but GitHub status check is hanging on _running_

seems to be some Azure pipelines glitch

**Comment by bruno-garcia** [comment]
Cool trick 😎

**Comment by bruno-garcia** [comment]
Thanks again @lmolkova
